### DISCUSSION TITLE
Is there anyway to specify custom asconfig.json path?

### DISCUSSION THREAD

**Discussion by ylazy** [BowlerHatLLC/vscode-as3mxml#778] [Q&A] [answered]
Hi!

I don't want to place asconfig.json at the root as it always gets pushed to the bottom of the folder.
Is there anyway to specify a custom path for asconfig.json, like the one bellow?

![image](https://github.com/user-attachments/assets/65944d89-36e1-4e55-81ad-525448e9ed45)

Thanks!

**Comment by joshtynjala** [comment, marked as answer]
Similar to TypeScript's tsconfig.json, asconfig.json is intended to be placed at the root of the project, and not in a sub-directory.

vscode-as3mxml can detect asconfig.json files in sub-directories, and it treats each one as a separate project. However, when asconfig.json is in a sub-directory, the sub-directory will be considered the root of the project for that asconfig.json file. You can certainly update the paths in asconfig.json (like _../src_ instead of _./src_ and _../bin_ instead of _./bin_). However, there may be some issues. For instance, if the workspace folder doesn't contain an asconfig.json at its root, VSCode won't know to activate the extension until you open an _.as_ or an _.mxml_ file in an editor. Even then, it may not initialize properly because the asconfig.json file isn't in one of the parent directories of the _.as_ and _.mxml_ files.
